src/plugins/kconfig/kconfig_delegate.cpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * @file
+ *
+ * @brief Delegate implementation for the `kconfig` plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include "kconfig_delegate.hpp"
+#include "file_utility.hpp"
+#include "kconfig_parser.hpp"
+#include <fstream>
+
+using kdb::Key;
+using kdb::KeySet;
+
+namespace elektra
+{
+using namespace kconfig;
+
+/**
+ * @brief This constructor creates a new delegate object used by the `kconfig` plugin
+ *
+ * @param config This key set contains configuration values provided by the `kconfig` plugin
+ */
+KconfigDelegate::KconfigDelegate (KeySet config)
+{
+	configuration = config;
+}
+
+/**
+ * @brief This method returns the configuration of the plugin, prefixing key names with the name of `parent`.
+ *
+ * @param parent This key specifies the name this function adds to the stored configuration values.
+ *
+ * @return A key set storing the configuration values of the plugin
+ */
+kdb::KeySet KconfigDelegate::getConfig (Key const & parent)
+{
+	KeySet keys{ 0, KS_END };
```

Comment:
```suggestion
	KeySet keys;
```

Reply:
should I change it in the `cpptemplate` plugin too?

Reply:
That sounds like a good idea. I should have done that myself, when I wrote that code 😊.